bids-validator/src/issues/list.test.ts

```diff
@@ -0,0 +1,65 @@
+import { assert, assertEquals } from '../deps/asserts.ts'
+import { loadSchema } from '../setup/loadSchema.ts'
+import { bidsIssues } from './list.ts'
+import { GenericSchema } from '../types/schema.ts'
+
+interface schemaError {
+  code: string
+  level?: string
+}
+
+function getSchemaErrors(
+  schema: GenericSchema,
+  rootSchema?: GenericSchema,
+  schemaPath?: string,
+): schemaError[] {
+  if (!rootSchema) {
+    rootSchema = schema
+  }
+  if (!schemaPath) {
+    schemaPath = 'schema.rules'
+  }
+  let errors = []
+  for (const key in schema) {
+    if (!(schema[key].constructor === Object)) {
+      continue
+    }
+    if (schema[key].constructor === Object) {
+      errors.push(...getSchemaErrors(
+        schema[key] as GenericSchema,
+        rootSchema,
+        `${schemaPath}.${key}`,
+      ))
+    }
+    let schemaEntry = schema[key] as object
+    if ('code' in schemaEntry && typeof schemaEntry['code'] === 'string') {
+      errors.push(schema[key])
+    }
+  }
+  return errors as schemaError[]
```

Comment:
Type `errors` at the head, iterate over entries instead of keys, and only check the type of constructor once:

```suggestion
  let errors: schemaError[] = []
  for (const [key, value] of Object.entries(schema)) {
    if (value.constructor !== Object) {
      continue
    }
    errors.push(...getSchemaErrors(
      value as GenericSchema,
      rootSchema,
      `${schemaPath}.${key}`,
    ))
    if ('code' in value && typeof value.code === 'string') {
      errors.push(value as unknown as schemaError)
    }
  }
  return errors
```